pkg/compute/kube/client.go

```diff
@@ -673,6 +673,22 @@ func (c *client) serviceToMeshServices(svc corev1.Service) []service.MeshService
 		// use port.appProtocol if specified, else use port protocol
 		meshSvc.Protocol = pointer.StringDeref(portSpec.AppProtocol, protocol)
 
+		// If app protocol was not detected and the protocol is TCP, use HTTP
+		// as the default app protocol.
+		// This is required for backward compatibility where users may be relying
+		// on implicit HTTP protocol selection.
+		//
+		// Note: it's possible for MeshService.Protocol to be unset for
+		// non TCP ports (eg. UDP) if we could not infer the corresponding
+		// app protocol via the appProtocol field or the port name.
+		// This will result in an error log for the specific port when the
+		// Envoy config is built for that port.
+		if meshSvc.Protocol == "" && portSpec.Protocol == corev1.ProtocolTCP {
+			log.Warn().Msgf("Did not detect app protocol for TCP port name=%s number=%d for service %s/%s, defaulting to http",
+				portSpec.Name, portSpec.Port, svc.Namespace, svc.Name)
+			meshSvc.Protocol = constants.ProtocolHTTP
+		}
+
```

Comment:
Should we log if the `meshSvc.Protocol` was never set or will this failure be seen later?

Reply:
This failure will be handled later where the protocol is actually used, so we can log it with enough context. Further, when we do add support for the currently unsupported protocols, we wouldn't need to update this code but rather the place where it logs an error.

Reply:
Okay, sounds good. Thanks for clarifying!